src/lib/bootloader/mbr_update.rb

```diff
@@ -166,21 +178,21 @@ def bios_id_for(device)
 
     # List of partition for disk that can be used for setting boot flag
     def activatable_partitions(disk)
-      partitions = target_map.fetch(disk, {}).fetch("partitions", [])
+      return [] unless disk
+
       # do not select swap and do not select BIOS grub partition
       # as it clear its special flags (bnc#894040)
-      partitions.select do |p|
-        p["used_fs"] != :swap && p["fsid"] != Yast::Partitions.fsid_bios_grub
+      devicegraph.disks.with(name: disk.name).partitions.reject do |part|
+        [Storage::ID_SWAP, Storage::ID_GPT_BIOS].include?(part.id)
       end
     end
 
-    def extended_partition_num(disk)
-      part = activatable_partitions(disk).find { |p| p["type"] == :extended }
+    def extended_partition(disk)
+      part = activatable_partitions(disk).find { |p| p.type == Storage::PartitionType_EXTENDED }
       return nil unless part
 
-      num = part["nr"]
-      log.info "Using extended partition #{num} instead"
-      num
+      log.info "Using extended partition #{part.number} instead"
```

Comment:
I am fine if youy use part.inspect instead. Especially when you do not return number, but object.